src/components/tabs/hooks/use-overflow-ref.ts

```diff
@@ -40,6 +40,14 @@ export default function useOverflowRef<T extends HTMLElement>(): [
         if (MutationObserver) {
           mutationObserver = new MutationObserver(requestOverflow)
           mutationObserver.observe(target, mutationOpts)
+          /**
+           * Handle when an overflow-ref-using element is rendered into tabs,
+           * observing the tab panel (which is expected to change visibility).
+           */
+          const targetParentNode = target.closest('[role="tabpanel"]')
+          if (targetParentNode) {
+            mutationObserver.observe(targetParentNode, mutationOpts)
+          }
```

Comment:
We discussed in a call, but for some additional context: This is sort of "tabs specific", but we decided that any use of this hook within a tab's content will need to be aware of of the visibility change and re-calculate accordingly, so it felt appropriate to bake that in here.